// webpack/config.js
const path = require('path');

module.exports = {
  // Entry point of the application
  entry: './src/index.js',  // Adjust the path to your main JavaScript file

  // Output configuration
  output: {
    filename: 'bundle.js',  // The name of the output file
    path: path.resolve(__dirname, 'dist'),  // The output directory
    publicPath: '/',  // Public URL of the output directory when referenced in a browser
  },

  // Module configuration
  module: {
    rules: [
      {
        test: /\.js$/,  // Apply this rule to .js files
        exclude: /node_modules/,  // Exclude the node_modules directory
        use: {
          loader: 'babel-loader',  // Use Babel to transpile JavaScript
          options: {
            presets: ['@babel/preset-env'],  // Use the preset for ES6+ features
          },
        },
      },
      {
        test: /\.css$/,  // Apply this rule to .css files
        use: ['style-loader', 'css-loader'],  // Use style-loader and css-loader
      },
      {
        test: /\.(png|jpg|gif|svg)$/,  // Apply this rule to image files
        use: [
          {
            loader: 'file-loader',
            options: {
              name: '[path][name].[ext]',  // Preserve the original file name and path
            },
          },
        ],
      },
    ],
  },

  // Development configuration
  devtool: 'source-map',  // Enable source maps for easier debugging

  // Development server configuration
  devServer: {
    contentBase: path.join(__dirname, 'dist'),  // Serve content from the dist directory
    compress: true,  // Enable gzip compression
    port: 9000,  // Port number for the dev server
    historyApiFallback: true,  // Enable support for HTML5 History API
  },

  // Plugins configuration
  plugins: [
    // Add any plugins you need here
  ],
};
